src/app/content/components/BookBanner.tsx

```diff
@@ -213,7 +215,7 @@ const BookBanner = () => {
     }
     e.preventDefault();
 
-    if (!await showConfirmation(services)) {
+    if (!await showConfirmation(services, dispatch, focsuedHighlight!)) {
```

Comment:
change `showConfirmation` to take `AppServices & MiddlewareAPI` as its first argument, that matches the hooks pattern, maybe also update `useServices` to return that to make life easier

Reply:
@TomWoodward I'm struggling with adding `MiddlwareAPI` to the context of `useServices`.

Reply:
[MiddlewareApi](https://github.com/reduxjs/redux/blob/a9181fc2f3230259dd36ba771bf6d2fb56081654/src/types/middleware.ts#L3) is just dispatch and getState, useServices is currently defined as:

```
export const useServices = () => React.useContext(servicesContext);   
```

you could change it to be:
```
export const useServices = () => ({
  ...React.useContext(servicesContext),
  dispatch: useDispatch(),
  getState: useStore().getState
```

or you could put that in a new hook, if you don't want to change the existing one